scipy/special/_logsumexp.py

```diff
@@ -202,7 +204,10 @@ def _logsumexp(a, b, axis, return_sign, xp):
     a_max, i_max = _elements_and_indices_with_max_real(a, axis=axis, xp=xp)
 
     # for precision, these terms are separated out of the main sum.
-    a[i_max] = -xp.inf
+    # TODO: we shouldn't be mutating in-place here unless we make a copy
+    # dask arrays do not copy before this somehow
+    #a[i_max] = -xp.inf
+    a = xp.where(i_max, -xp.asarray(xp.inf, dtype=a.dtype), a)
```

Comment:
`i_max` is a boolean mask, so this can be changed unconditionally to `where` I think, according to the rule of thumb created after benchmarking in https://github.com/scipy/scipy/issues/22246#issuecomment-2574224833.

Reply:
Thanks for the catch, I should also clean up the comment since that's outdated and now longer applies.

Reply:
This has the potential for performance degradation in numpy et al; needs benchmarking to prove otherwise